signal-hook-tokio/src/lib.rs

```diff
@@ -75,6 +76,9 @@ pub mod v0_1 {
     use tokio_0_1::io::AsyncRead;
     use tokio_0_1::net::unix::UnixStream;
 
+    /// A shareable handle to control an associated [`Signals`] instance.
+    pub type ControllerHandle = Arc<Controller<UnixStream>>;
```

Comment:
This was my first thought, too. Then I thought it would be nice to show users that cloning is very cheap by not hiding the Arc. But I don't have a strong opinion on that.

Reply:
Having seen some libraries, things called `Handle` are usually cheap to clone. It is probably worth mentioning in the documentation comment too, though.

All in all, I'd prefer to keep the exact internal representation hidden, for future flexibility of implementation.